### ISSUE TITLE
How do I get the caret to appear inside a custom text-and-image-combined Node?

### ISSUE THREAD

**Issue by drinkitup777** [FXMisc/RichTextFX#949]
I'm trying to display a custom node in my subclass of `GenericStyledArea`, a syntax-highlighted markdown editor that displays images inline. I want the final result of my custom node to look something like this:

[![example.png](https://i.postimg.cc/pXDm66Hz/example.png)](https://postimg.cc/7fZPCXbZ)

My goal is to create a custom node that displays a markdown image link and a preview of the image above it, but also have the text below remain selectable and editable by the user. But my problem is that I can't get the caret to appear inside my custom node.

How do I make the caret appear in the text of my custom node?

---

My approach has been to bundle a `TextExt` together with an `ImageView` inside a `VBox`, and return that from my node factory, but now the caret will no longer enter the text in my custom node. It seems returning anything other than a `TextExt` from the node factory makes the caret do this.

Just for clarity, this is what I'm returning from my node factory:

```java
    // **** this is an shortened version just for demo purposes; my real code does roughly the same thing ****

    private static Node createNode(StyledSegment<Either<String, Embeddable>, Long> styledSegment) {
        Either<String, Embeddable> segment = styledSegment.getSegment();

        if (segment.isLeft()) {
            return StyledTextArea.createStyledTextNode(segment.getLeft(), styledSegment.getStyle(),
                    (text, styles) -> text.getStyleClass().addAll(Style.toCssClasses(styles)));
        } else if (segment.isRight()) {
            VBox box = new VBox();

            TextExt t = new TextExt();
            t.setText(this.getRawText());
            t.setTextOrigin(VPos.TOP);
            t.getStyleClass().add("hyperlink");

            ImageView view = this.toImageView();

            box.getChildren().add(view);
            box.getChildren().add(t);

            return box;
        }

        return null;
    }
```

**Comment by Jugen** [comment]
Yeah, `It seems returning anything other than a TextExt from the node factory makes the caret do this`, that's right. Custom objects have a width of one and no drill down is done to see if there's any Text inside it, so the caret won't enter into it.

Alternatives may be to either make the TextExt object a TextField (or RichTextField) and CSS it to remove the field border, or to make it a Label and listen for clicks on it to enter edit mode.
